_docs/030-getting-started-docker-for-windows.md

```diff
@@ -178,6 +178,64 @@ curl http://localhost:31380/ -H 'Host: square.default.example.com' -H 'Content-T
 64
 ```
 
+## create a function from code in a local directory
+
+You can use riff to build functions from source in a local directory, instead of first committing the code to a repo on GitHub.
+
+For this to work with Docker Hub from Windows, a small workaround is required to support the multiple ways the Docker Hub registry can be named.
+
+Create a temporary file called `dockerhub.json` using your own docker credentials
```

Comment:
We could avoid the temp file by using:

```
echo '{"ServerURL": "https://index.docker.io", "Username": "<DOCKER-ID>", "Secret": "<PASSWORD>"}' | docker-credential-wincred store
```

Reply:
Is there a way to prompt for the pw in PowerShell to avoid leaking it into the history?

Reply:
This might work:

```
$dockerid = Read-Host "Please enter your Docker ID: "
$password = Read-Host -assecurestring "Please enter your password: "
echo $('{"ServerURL": "https://index.docker.io", "Username": "' +  $dockerid + '", "Secret": "' + [System.Runtime.InteropServices.Marshal]::PtrToStringAuto([System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($password)) + '"}') | docker-credential-wincred store
```

Reply:
nice - i'll test it tomrrow - thanks!

Reply:
trying this out now.

Reply:
thanks - using your (hopefully) safer method.